nautobot/dcim/form_mixins.py

```diff
@@ -0,0 +1,140 @@
+from django import forms
+from django.contrib.contenttypes.models import ContentType
+
+from nautobot.dcim.models import Region, Site, Location
+from nautobot.utilities.forms import CSVModelChoiceField, DynamicModelChoiceField, DynamicModelMultipleChoiceField
+
+
+class LocatableModelFormMixin(forms.Form):
+    """
+    Mixin for model forms that can link to a Site (filtered by Region) and/or Location.
+
+    In the long term when Region and Site are collapsed into Location this should greatly reduce the number
+    of distinct places in the code that we need to touch.
+    """
+
+    region = DynamicModelChoiceField(
+        queryset=Region.objects.all(),
+        required=False,
+        initial_params={"sites": "$site"},
+    )
+    site = DynamicModelChoiceField(
+        queryset=Site.objects.all(),
+        query_params={"region_id": "$region"},
+    )
+    location = DynamicModelChoiceField(
+        queryset=Location.objects.all(),
+        required=False,
+        # TODO: query_params={"base_site": "$site"},
```

Comment:
Yeah, as I noted in the PR description:

> One thing I haven't yet figured out if it's possible/feasible is to make the Location drop-down self-filter based on the selected Site value, similar to how we currently filter the Site dropdown based on the selected Region. It'd be pretty straightforward to filter to the root locations under a given site, but I don't have a good idea yet on how to also include all descendant locations of these.

`base_site` isn't a database field so it would have to be a custom method filter, and I'm concerned about performance impacts thereof.  I noted in https://github.com/nautobot/nautobot/issues/510#issuecomment-1168762953 that this is relatively inexpensive with django-mptt (it'd be something like `Location.objects.filter(site=base_site).get_descendants(include_self=True)` but I don't see a similar pattern currently supported in django-tree-queries. This may be a factor in favor of switching back to django-mptt for 1.4.0 final.

Reply:
Custom method filters for a specific use case here (and limited since we will eventually remove Site and Region in the future) should be fine enough for now.

No sense in making perfection the enemy of good here.